### ISSUE TITLE
It's not possible to modify some tables

### ISSUE THREAD

**Issue by gozzoo** [sqlitebrowser/sqlitebrowser#1733]
Details for the issue
--------------------

#### What did you do?
I opened a database with few existing tables. All tables did have multiple fields

#### What did you expect to see?

![image](https://user-images.githubusercontent.com/296970/52505593-aa425800-2bf4-11e9-87fa-4c16512e69ae.png)

This is with version 3.10.1

#### What did you see instead?
For some tables there is no expand icon.

![image](https://user-images.githubusercontent.com/296970/52497799-a952fb80-2bdf-11e9-9abd-20c5c21918c9.png)

When I right-click the name of such table and select 'Modify table' I see defintion for an empty table.

![image](https://user-images.githubusercontent.com/296970/52497931-0d75bf80-2be0-11e9-925f-bb43f6681c5e.png)

Useful extra information
-------------------------

The info below often helps, please fill it out if you're able to. :)

#### What operating system are you using?

- [x] Windows: (Windows 7 32 bit)

#### What is your DB4S version?

- [x] 3.11.0

#### Did you also

- [ ] Try out the latest nightly version: https://github.com/sqlitebrowser/sqlitebrowser#nightly-builds
- [ ] Search for an existing similar issue: https://github.com/sqlitebrowser/sqlitebrowser/issues?utf8=%E2%9C%93&q=is%3Aissue%20

**Comment by chrisjlocke** [comment]
The schema for those tables might contain some weird characters?  Is it possible to email me the database?  db4s [at] chrisjlocke [dot] co [dot] uk.  You can send me a database sans data (executing the necessary 'delete from ...' commands from the 'execute sql' tab should still work, regardless of whether its fully read the schema)

**Comment by gozzoo** [comment]
I updated my report. It works fine with version 3.10.1

Here is the table deifnition:

```
CREATE TABLE "auto_measurement" (
	`id`	INTEGER NOT NULL,
	`full`	BOOLEAN,
	`type`	INTEGER,
	`entry`	INTEGER DEFAULT (DATETIME(CURRENT_TIMESTAMP,'LOCALTIME')),
	`exit`	INTEGER,
	`plate`	char ( 50 ),
	`seller`	CHAR ( 50 ),
	`buyer`	CHAR ( 50 ),
	`driver`	CHAR ( 50 ),
	`storage`	TEXT,
	`storage2`	TEXT,
	`trailer`	CHAR ( 50 ),
	`product`	TEXT,
	`product_code`	TEXT,
	`tara`	NUMERIC,
	`bruto`	NUMERIC,
	`neto`	NUMERIC,
	`discount`	NUMERIC DEFAULT 0,
	`price_kg`	NUMERIC,
	`note`	TEXT,
	`carrier`	char ( 50 ),
	`subcontractor`	char ( 50 ),
	`site`	INTEGER,
	`user1`	TEXT,
	`user2`	TEXT,
	`user3`	TEXT,
	`operator_in`	TEXT,
	`operator_out`	TEXT,
	`status`	INTEGER,
	`scale`	TEXT,
	`manual`	INTEGER,
	`price`	NUMERIC,
	PRIMARY KEY(`id`)
);
```

**Comment by chrisjlocke** [comment]
I assume you're aware thats not a SQLite compatible table definition.  SQLite doesn't understand what 'char' is, or 'boolean'.  It guesses, but it certainly isn't going to respect the 50 character field limits.  I'm guessing its the 'entry' line causing grief - if you remove that, it works fine.  It does appear valid ... its just our parser can't parse it.  Doh.

**Comment by justinclift** [comment]
@gozzoo Out of curiosity, which database dialect is that SQL table definition created for?

**Comment by gozzoo** [comment]
@justinclift No specific dialect in particular. I was expecting our projectt needs to outgrow sqlite pretty soon and wanted the database scripts to be portable. I never experimented running them on another dabtabase.

**Comment by MKleusberg** [comment]
This is a very weird bug. Looks like it was introduced by adding some more rules to our grammar parser. And even when these rules aren't used anywhere they cause some problems. I have disabled them now which shouldn't cause any problems since we don't use them (yet). So with the next nightly build everything should work as expected again.

Can you download the nightly build tomorrow and double check, @gozzoo? :smile:  If it's working again in the nightly this is definitely a regression from 3.10.1 and the fix should definitely be included in a 3.11.1 release :wink:

**Comment by justinclift** [comment]
That reminds me... the build server has been off all day, so the nightlies haven't run yet.  I'll manually kick off the build scripts now.

Also just created a 3.11.1 milestone, so we have something to add this to.  Made the due date two weeks from now, but we can adjust that sooner/later if need be. :smile:

**Comment by justinclift** [comment]
@gozzoo The Win64 nightly builds just finished, so they're online and ready for testing if/when you have a chance:

&nbsp; &nbsp; https://nightlies.sqlitebrowser.org/latest/

**Comment by gozzoo** [comment]
It's working fine now, thanks.

**Comment by chrisjlocke** [comment]
Thanks for letting us know.  Will close this as its resolved (of sorts).

**Comment by MKleusberg** [comment]
I've cherry-picked this over to the 3.11.x branch so it will be included in the 3.11.1 release :smile: Thanks again for pointing this out, @gozzoo!

**Comment by gozzoo** [comment]
Thanks @MKleusberg
